### PR TITLE
langchain-core 0.3.21 ❄️

### PR DESCRIPTION
**Pull request by lorepirri**
langchain-core 0.3.21 :snowflake:

**Destination channel:** Snowflake

### Links

- [PKG-6046]
- dev_url (pypi): https://github.com/langchain-ai/langchain/tree/langchain-core==0.3.21/libs/core
- conda-forge: https://github.com/conda-forge/langchain-core-feedstock/blob/main/recipe/meta.yaml
- pypi: https://pypi.org/project/langchain-core/0.3.21
- pypi inspector: https://inspector.pypi.io/project/langchain-core/0.3.21

### Depends on

- AnacondaRecipes/pytest-mock-feedstock#3

### Explanation of changes:

- bump version
- refactor tests section
- rebuild `pytest-mock` to enable all tests

[PKG-6046]: https://anaconda.atlassian.net/browse/PKG-6046?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,76 +1,82 @@
...
+  source_files:
+    - libs/core/tests/unit_tests
+    - libs/core/tests/__init__.py
   requires:
     - pip
-    - pytest >=7.3.0,<8
-    - pytest-mock >=3.10.0,<4  # [not (osx and x86 and py>=312)]
-    - numpy >=1.24.0,<2
-    - responses >=0.25.0,<0.26.0
+    - pytest >=7.3.0
```
Comment (skupr-anaconda) [inline recipe/meta.yaml:66]:
```suggestion
    - pytest >=8.0.0,<9.0.0
```
https://github.com/langchain-ai/langchain/blob/langchain-core%3D%3D0.3.19/libs/core/pyproject.toml#L105

File: recipe/meta.yaml [hunk 2]
```diff
@@ -1,76 +1,82 @@
...
+    - freezegun #>=1.2.2,<2
+    - pytest-mock >=3.10.0,<4
     # Versions expected by upstream not on defaults yet.
     - pytest-asyncio #>=0.21.1,<0.22
-    - freezegun #>=1.2.2,<2
+    - responses >=0.25.0,<0.26.0
     # Deps below not on defaults, required for ignored tests below
     # syrupy, grandalf, pytest-profiling, pytest-watcher
```
Comment (skupr-anaconda) [inline recipe/meta.yaml]:
```suggestion
    # syrupy, grandalf, pytest-profiling, pytest-watcher, pytest-socket
```
https://github.com/langchain-ai/langchain/blob/langchain-core%3D%3D0.3.19/libs/core/pyproject.toml#L113-L113C14

**Review by skupr-anaconda — APPROVED** [review]
LGTM but minor fixes

File: recipe/meta.yaml (hunk 1 above)
Comment (skupr-anaconda) [reply recipe/meta.yaml:66]: `nothing provides requested pytest >=8.0.0,<9.0.0`
so revert the previous pin` >=7.3.0`. I think it's fine too

**Comment by skupr-anaconda** [comment]
@lorepirri @cbouss The ticket https://anaconda.atlassian.net/browse/PKG-6022 says that the target channel is `main`. Is it correct? I had the same channel change in https://github.com/AnacondaRecipes/langsmith-feedstock/pull/6

**Comment by cbouss** [comment]
> https://anaconda.atlassian.net/browse/PKG-6022

The package is transitioning to main. This ticket is for this sprint, the other ticket for next sprint.

**Comment by skupr-anaconda** [comment]
> > https://anaconda.atlassian.net/browse/PKG-6022
> 
> The package is transitioning to main. This ticket is for this sprint, the other ticket for next sprint.

Thanks for the clarification!

**Review by boldorider4 — APPROVED** [review]
LGTM

File: recipe/meta.yaml (hunk 1 above)
Comment (lorepirri) [reply recipe/meta.yaml:66]: reverted

**Pull request merged**